internal/tracker/tracker.go

```
@@ -0,0 +1,194 @@
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not create sqlite connection to %s", path)
+	}
+
+	err = t.ensureTablesExist()
+	if err != nil {
+		return nil, errs.Wrap(err, "Could not set up database")
+	}
+
+	return t, nil
+}
+
+func (t *Tracker) ensureTablesExist() error {
+	for _, trackable := range []TrackingType{Files, Directories, FileTag, Environment} {
+		_, err := t.db.Exec(fmt.Sprintf("CREATE TABLE IF NOT EXISTS %s (key string NOT NULL PRIMARY KEY, value text)", trackable))
+		if err != nil {
+			return errs.Wrap(err, "Could not create files table in tracker database")
+		}
+	}
+
+	return nil
+}
+
+func (t *Tracker) Close() error {
+	if t.closed {
+		return nil
+	}
+	t.closed = true
+	return t.db.Close()
+}
+
+func (t *Tracker) Track(ts ...Trackable) error {
+	for _, tr := range ts {
+		err := tr.Store(t.db)
+		if err != nil {
+			return errs.Wrap(err, "Could not store trackable")
+		}
+	}
+	return nil
+}
+
+func (t *Tracker) GetFiles() ([]string, error) {
+	return t.getStringSlice(Files)
+}
+
+func (t *Tracker) GetFile(key string) (string, error) {
+	return t.getString(Files, key)
+}
+
+func (t *Tracker) GetDirectories() ([]string, error) {
+	return t.getStringSlice(Directories)
+}
+
+func (t *Tracker) GetDirectory(key string) (string, error) {
+	return t.getString(Directories, key)
+}
+
+func (t *Tracker) GetFileTags() ([]string, error) {
+	return t.getStringSlice(FileTag)
+}
+
+func (t *Tracker) GetFileTag(key string) (string, error) {
+	return t.getString(FileTag, key)
+}
+
+func (t *Tracker) GetEnvironmentVariables() (map[string]string, error) {
+	rows, err := t.db.Query(fmt.Sprintf("SELECT key, value FROM %s", Environment))
+	if err != nil {
+		return nil, errs.Wrap(err, "Get environment variables query failed")
+	}
+
+	env := make(map[string]string)
+	for rows.Next() {
+		var key, value string
+		err := rows.Scan(&key, &value)
+		if err != nil {
+			logging.Error("Failed to scan path value: %v", err)
```

Should return this error?